imcsdk/apis/admin/ntp.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2016 Cisco Systems, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+This module implements all the ntp related functionality
+"""
+from imcsdk.mometa.comm.CommNtpProvider import CommNtpProvider
+from imcsdk.imccoreutils import _is_valid_arg
+
+
+def _get_ntp_mo(handle=None, from_server=False):
+    from imcsdk.imcexception import ImcOperationError
+
+    mo = CommNtpProvider(parent_mo_or_dn="sys/svc-ext")
+    if from_server:
+        if handle is None:
+            raise ImcOperationError("Get NTP Settings", "Handle is None")
+
+        mo = handle.query_dn(mo.dn)
+        if mo is None:
+            raise ImcOperationError("Get NTP Settings", "MO doesn't exist")
+    return mo
+
+
+def _set_ntp_servers(mo, ntp_servers):
+    args = {}
+    for server in ntp_servers:
+        key = "ntp_server" + str(server["id"])
+        args[key] = server["ip"]
+    mo.set_prop_multiple(**args)
+
+
+def ntp_enable(handle, ntp_servers=[]):
+    """
+    Enable NTP
+
+    Args:
+        handle (ImcHandle)
+        ntp_servers (list): List of dictionaries of the type
+                            [{"id": 1, "ip": "192.168.1.1"},
+                             {"id": 2, "ip": "192.168.1.2"}]
+                            Upto 4 ntp servers can be specified.
+
+    Returns:
+        CommNtpProvider object
+
+    Example:
+        ntp_enable(handle,
+                   ntp_servers = [{"id": 1, "ip": "192.168.1.1"},
+                                  {"id": 2, "ip": "192.168.1.2"}]
+    """
+
+    mo = _get_ntp_mo()
+    mo.ntp_enable = "yes"
+
+    _set_ntp_servers(mo, ntp_servers)
```

Comment:
@waghswapnil ok, but I feel we should fail it. Because the user should not assume 5 entries have got programmed.. We should accept only <=4